tests/foreman/sys/test_katello_certs_check.py

```diff
@@ -73,11 +73,12 @@ class TestKatelloCertsCheck:
     CA cert (a.k.a cacert.crt or rootCA.pem) can be used as bundle file.
     """
 
+    error_message = f"error 26 at 0 depth lookup: {'unsupported' if settings.repos.rhel_major_version == '8' else 'unsuitable'} certificate purpose"
```

Comment:
This error surprises me a bit. If we don't assert the explicit message we emit ourselves:
https://github.com/theforeman/foreman-installer/blob/7c2b54267a933bcf0261396c0950d51c8d5bec5c/bin/katello-certs-check#L157-L169

Perhaps that's not needed and we can simply assert the right code path is used by checking for exit code 4?

Reply:
This is what I see. I jumped the gun and assumed you suggested to use `The /root/cacert.crt does not verify the /root/certs/invalid.crt`. I see you meant check of error code 4 but I don't see it in the logs.
```
    
    Checking CA bundle against the certificate file: 
    [FAIL]
    
    The /root/cacert.crt does not verify the /root/certs/invalid.crt
    CN=foreman.example.com
    error 20 at 0 depth lookup: unable to get local issuer certificate
    error /root/certs/invalid.crt: verification failed
    
    Checking CA bundle size: 1
    [OK]
    
    Checking if CA bundle has trust rules: 0
    [OK]
    
    Checking Subject Alt Name on certificate 
    [FAIL]
    
    The /root/certs/invalid.crt does not contain a Subject Alt Name. Common Name is deprecated, use Subject Alt Name instead. See: https://tools.ietf.org/html/rfc2818#section-3.1
    Checking Key Usage extension on certificate for Key Encipherment 
    [OK]
    
    Checking for use of shortname as CN
    
    stderr:
    
    status: 1
```

Reply:
That signals something is broken. The error function sets the exit code:
https://github.com/theforeman/foreman-installer/blob/7c2b54267a933bcf0261396c0950d51c8d5bec5c/bin/katello-certs-check#L65-L70

Then It's also surprising the command exits unexpectedly, without showing `[OK]` or `[FAIL]`. That looks like `check-shortname` isn't following an expected code path:
https://github.com/theforeman/foreman-installer/blob/7c2b54267a933bcf0261396c0950d51c8d5bec5c/bin/katello-certs-check#L229-L250

Given there was already a warning that it doesn't contain a SAN I think we can conclude it's missing an `else` statement with `success`.

Reply:
So it's a bug😅. DO you want me to file an issue?

Reply:
It would be great to have it covered in our tests: https://github.com/theforeman/foreman-installer/blob/develop/spec/katello_certs_check_spec.rb is what we use in CI.

Do I read it well that these certs are generated using `tests/foreman/data/certs.sh`? In particular:
https://github.com/SatelliteQE/robottelo/blob/fdad16a5ae1e7a5f7cda641b8f27d6ab070c2fd6/tests/foreman/data/certs.sh#L48-L49

Reply:
yes